src/components/signupForm/SignUpForm.tsx

```diff
@@ -54,61 +60,107 @@ function SignupForm({ onSubmit }: SignupFormProps) {
     }
     setEmailError('');
     setPasswordError('');
-    onSubmit(username, email, password, confirmPassword);
+    try {
+      const response = await fetch(APIBASE + 'auth/signup', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        credentials: 'include',
+        body: JSON.stringify({
```

Comment:
**Backend is currently unable to read username because username field in JSON payload is incorrectly labeled**

Change 
```typescript
body: JSON.stringify({
  // ... rest of payload
  inputUsername,
})
```
to 
```typescript
body: JSON.stringify({
    // rest of payload
  username: inputUsername, 
})
```

You could also just change inputUsername variable to have the name username, which is likely the cleaner fix

Reply:
That's what I get for find and replace all username with inputUsername... Had to differentiate from authContext username.